provider/pkg/provider/provider.go

```diff
@@ -398,32 +403,82 @@ func (k *azureNativeProvider) Invoke(ctx context.Context, req *rpc.InvokeRequest
 	return &rpc.InvokeResponse{Return: result}, nil
 }
 
-func (k *azureNativeProvider) getClientToken(ctx context.Context, authConfig *authConfig, endpointArg resource.PropertyValue) (string, error) {
-	endpoint := k.tokenEndpoint(endpointArg)
+func (k *azureNativeProvider) getClientConfig(ctx context.Context) (config *ClientConfig, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			config = nil
+			err = fmt.Errorf("panic recovered in getClientConfig: %v", r)
+		}
+	}()
 
-	if util.EnableAzcoreBackend() {
-		cred, err := k.newTokenCredential()
+	if !util.EnableAzcoreBackend() {
+		auth, err := k.getAuthConfig()
```

Comment:
Nit: handle the fall through case first to reduce nesting.
```suggestion
	if util.EnableAzcoreBackend() {
		GetClientConfig(ctx, &k.authConfig, k.credential)
	}
	
	auth, err := k.getAuthConfig()
	...
```